### PR TITLE
Fedora 34

### PR DESCRIPTION
**Pull request by KKoukiou**
- [x] Move our fedora-33 image to cloud first, and fix tests: https://github.com/cockpit-project/bots/pull/1719 (as a lot of failures are due to that)
 - [x] `test/verify/check-kdump TestKdump.testBasic`: [bugzilla](https://bugzilla.redhat.com/show_bug.cgi?id=1933970), known issue https://github.com/cockpit-project/bots/issues/1734 , needs naughty pattern
 - [x] `test/verify/check-static-login TestLogin.testBasic` root bridge core dump -- *update*: pitti can't reproduce any more, likely fixed with the utmp binary fix below. Will re-open if CI disagrees.
 - [x]  test/verify/check-system-journal TestJournal.testBasic systemd changed log message https://github.com/systemd/systemd/commit/f5ec78e503212855eb2adf92628023148afcea6f
 - [x]  test/verify/check-shell-active-pages, check-systemd-terminal fail because of failed unit appearing on the first terminal line
 - [x]  test/verify/check-users TestAccounts.test* parsing utmp closes the cockpit channel - passing the binary: true option fixes the issue
 - [x] check-realms realm join failure for AD: [bugzilla](https://bugzilla.redhat.com/show_bug.cgi?id=1934124), known issue https://github.com/cockpit-project/bots/issues/1735 : https://github.com/cockpit-project/bots/pull/1740
 - [x] `TestStorageFormat.testFormatTypes` failure is https://github.com/cockpit-project/bots/issues/1733 , needs applying the naughty to fedora-34 as well (dosfstools-4.2-1.fc34.x86_64)
 - [x] `TestCurrentMetrics.testMemory`: fixed in PR #15454
 - [x]  check-storage-resize TestStorageResize.testResizeLuks: direct issue from https://bugzilla.redhat.com/show_bug.cgi?id=1934567, fixed locally
 - [x] fix watching binary channels: PR #15467

### CODE REVIEW

**Comment by martinpitt** [comment]
Re-running after PR #15406, to see how much that helps. [previous run](https://logs-https-frontdoor.apps.ocp.ci.centos.org/logs/pull-15350-20210216-154012-9e85cdb2-fedora-34-bots-1685/log.html) had 57 failures

**Comment by martinpitt** [comment]
A lot of tests fail on this:
```
audit: type=1400 audit(1614158730.147:8617): avc:  denied  { integrity } for  pid=512 comm="systemd-logind" lockdown_reason="hibernation" scontext=system_u:system_r:systemd_logind_t:s0 tcontext=system_u:system_r:systemd_logind_t:s0 tclass=lockdown permissive=0
```
It's not reported to bugzilla yet. Let's first retry this with the new image from https://github.com/cockpit-project/bots/pull/1685 before we do, though.

**Comment by martinpitt** [comment]
Argh, it's conflict-y now, needs rebase.

**Comment by martinpitt** [comment]
Well, 57 → 25 failures, that's progress :grin:

**Comment by martinpitt** [comment]
Running against https://github.com/cockpit-project/bots/pull/1718 which buils fedora-34 from the fedora 34 instead of rawhide cloud image.

**Comment by martinpitt** [comment]
Let's block this on https://github.com/cockpit-project/bots/pull/1719 and not waste work on this until then.

**Comment by martinpitt** [comment]
Re-running against https://github.com/cockpit-project/bots/pull/1719 which now has firewalld. This ought to fix, or at least improve, all the [networking failures](https://logs.cockpit-project.org/logs/pull-15350-20210225-065606-ae568063-fedora-34-bots-1718/log.html)

**Comment by martinpitt** [comment]
Networking tests are ok now, down to 23.

**Comment by martinpitt** [comment]
I pushed a rebase, as the previous f34 run never finished.

**Comment by martinpitt** [comment]
I have started to investigate failures here and make a list of TODOs in the description. Let's please all use this to track remaining issues. I reported the kdump failure downstream and filed a known issue.

**Comment by martinpitt** [comment]
I pushed two naughties for kdump and realmd to https://github.com/cockpit-project/bots/pull/1719 , re-running.

File: test/common/testlib.py
Comment (martinpitt) [inline test/common/testlib.py:1121]: Please add a comment to point to bugzilla.

File: test/verify/check-system-journal
Comment (martinpitt) [inline test/verify/check-system-journal:235]: This needs to accept both variants for a while.

**Review by martinpitt — CHANGES_REQUESTED** [review]
A few things to fix, but I think we should land this first batch as soon as possible. Especially the utmp fix. So I suggest on next push you drop the no-test label, and then we can look at the remaining failures separately. Epharisto!

File: test/verify/check-system-journal
Comment (martinpitt) [reply test/verify/check-system-journal]: Fixed.

File: test/common/testlib.py
Comment (martinpitt) [reply test/common/testlib.py:1122]: Fixed.

**Comment by martinpitt** [comment]
Some notes about the [TestAccounts.testBasic](https://logs.cockpit-project.org/logs/pull-15350-20210304-085507-d4f84fcb-ubuntu-2004/log.html#270) failure:
 * Watching the old way (text files) always called the watch callback with `(null, null, "protocol-error")`.
 * Watching the file in binary mode is conceptually right, but the callback then never actually gets called. While `stat /var/run/utmp` confirms that the file is updated , and `diff` confirms that the content changes.

So watching binary files is indeed broken.

**Comment by martinpitt** [comment]
I filed PR #15467 to fix binary file watching, and there I also added the users page fix to demonstrate that it works in real life. After that lands, let's rebase this and land this PR.

File: test/verify/check-storage-resize
Comment (martinpitt) [inline test/verify/check-storage-resize:96]: Wouldn't that be better as a naughty?

Comment (KKoukiou) [reply test/verify/check-storage-resize:96]: The naughty will make the test fail premature (xfail of course). With this small Workaround we make sure that the test can finish, and that the rest of functionality is actually ok.
 
If you disagree I am happy to extend the existing naughty.

Lastly, when this issue is fixed, the Mount button will be missing so this will start failing. In this way we will get naughtified and we will revert the HACK.

Comment (KKoukiou) [reply test/verify/check-storage-resize:96]: I will create a naughty

Comment (martinpitt) [reply test/verify/check-storage-resize]: If it skips too much functionality which is otherwise working, that's ok -- I mostly was interested in the rationale. Thanks!

Comment (KKoukiou) [reply test/verify/check-storage-resize]: https://github.com/cockpit-project/bots/pull/1746 contains already the naughty.

**Review by martinpitt — APPROVED** [review]
All tests except for 34 are green, so fine for me to land and then investigate the remaining issues separately. Thanks, really great work!

**Comment by martinpitt** [comment]
The [last run](https://logs.cockpit-project.org/logs/pull-15350-20210304-121847-1a728067-fedora-34-bots-1746/log.html) now looks unexpectedly sad. Retrying to have a comparison.

**Pull request merged**
